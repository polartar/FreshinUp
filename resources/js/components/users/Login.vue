<template>
  <v-form @submit.prevent="save()">
    <v-card class="elevation-12 rounded-card">
      <v-toolbar
        flat
        color="transparent"
      >
        <v-img
          :src="logo"
          height="64"
          contain
        />
      </v-toolbar>

      <v-card-title
        primary-title
        class="justify-center"
      >
        <h3
          class="headline mb-0 primary--text font-weight-bold"
          :style="{ justifyContent: 'center'}"
        >
          Log in to {{ title }}
        </h3>
      </v-card-title>

      <v-card-text>
        <v-text-field
          v-model="email"
          name="login"
          placeholder="Email address"
          type="text"
          label="Email address"
        />
        <v-text-field
          id="password"
          v-model="password"
          name="password"
          placeholder="Password"
          label="Password"
          type="password"
        />
      </v-card-text>

      <v-card-actions>
        <v-layout
          row
          wrap
        >
          <v-flex xs12>
            Forgot password ?
            <v-btn
              class="text-uppercase"
              color="primary"
              type="button"
              flat
              small
              @click="passwordForgot"
            >
              Reset password here
            </v-btn>
          </v-flex>
          <v-flex xs12>
            <v-btn
              color="primary"
              class="ff-login__button mt-2 mb-2"
              type="submit"
              :loading="isLoading"
            >
              Login
            </v-btn>
          </v-flex>
          <v-flex>
            <p class="mb-1 mt-2">
              Not a member ?
            </p>
            <p>
              Sign up as a
              <a
                href="#register-customer"
                @click.prevent="registerAs('customer')"
              >Customer</a>
              or
              <a
                href="#register-supplier"
                @click.prevent="registerAs('supplier')"
              >Join our fleet</a>
            </p>
          </v-flex>
        </v-layout>
      </v-card-actions>
    </v-card>
    <slot />
  </v-form>
</template>

<script>
// TODO: move to core-ui
import MapValueKeysToData from '~/mixins/MapValueKeysToData'

export const DEFAULT_VALUE = {
  email: '',
  password: ''
}
export default {
  mixins: [MapValueKeysToData],
  props: {
    value: { type: Object, default: () => DEFAULT_VALUE },
    logo: { type: String },
    title: { type: String },
    isLoading: { type: Boolean, default: false }
  },
  data () {
    return {
      ...DEFAULT_VALUE
    }
  },
  methods: {
    registerAs (type) {
      this.$emit('register-as', type)
    },
    passwordForgot () {
      this.$emit('password-forgot')
    }
  }
}
</script>
