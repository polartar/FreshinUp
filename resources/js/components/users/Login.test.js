import { mount, shallowMount } from '@vue/test-utils'
import Component, { DEFAULT_VALUE } from '~/components/users/Login.vue'
import * as Stories from '~/components/users/Login.stories'

describe('components/users/Login', () => {
  describe('Snapshots', () => {
    it('Basic', async () => {
      const wrapper = mount(Stories.Basic())
      expect(wrapper.element).toMatchSnapshot()
    })
    it('Loading', async () => {
      const wrapper = mount(Stories.Loading())
      expect(wrapper.element).toMatchSnapshot()
    })
    it('WithError', async () => {
      const wrapper = mount(Stories.WithError())
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Props & Computed', () => {
    test('logo', () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.logo).not.toBeDefined()

      wrapper.setProps({
        logo: '/images/logo.png'
      })
      expect(wrapper.vm.logo).toEqual('/images/logo.png')
    })
    test('title', () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.title).not.toBeDefined()

      const title = 'FoodFleet'
      wrapper.setProps({
        title
      })
      expect(wrapper.vm.title).toEqual(title)
    })
    test('isLoading', () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.vm.isLoading).toEqual(false)

      wrapper.setProps({
        isLoading: true
      })
      expect(wrapper.vm.isLoading).toEqual(true)
    })
  })

  describe('Methods', () => {
    describe('save()', () => {
      test('default', () => {
        const wrapper = mount(Component)
        wrapper.vm.save()
        let emitted = wrapper.emitted().input
        expect(emitted).toBeTruthy()
        expect(emitted[0][0]).toMatchObject(DEFAULT_VALUE)
      })
      test('set', async () => {
        const wrapper = mount(Component)
        const payload = {
          email: 'john.doe@domain.fr',
          password: 'pass123'
        }
        wrapper.setProps({
          value: payload
        })
        await wrapper.vm.$nextTick()
        wrapper.vm.save()
        const emitted = wrapper.emitted().input
        expect(emitted).toBeTruthy()
        expect(emitted[0][0]).toMatchObject(payload)
      })
    })

    test('passwordForgot()', () => {
      const wrapper = mount(Component)
      wrapper.vm.passwordForgot()
      let emitted = wrapper.emitted()['password-forgot']
      expect(emitted).toBeTruthy()
    })

    test('registerAs(type)', () => {
      const wrapper = mount(Component)
      wrapper.vm.registerAs('supplier')
      let emitted = wrapper.emitted()['register-as']
      expect(emitted).toBeTruthy()
      expect(emitted[0][0]).toEqual('supplier')
    })
  })
})
