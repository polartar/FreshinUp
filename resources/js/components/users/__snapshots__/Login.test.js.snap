// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/users/Login Snapshots Basic 1`] = `
<div
  class="container fluid"
>
  <div
    class="layout align-center justify-center"
  >
    <div
      class="flex xs12 sm10 md4"
    >
      <form
        class="v-form"
        novalidate="novalidate"
      >
        <div
          class="elevation-12 rounded-card v-card v-sheet theme--light"
        >
          <nav
            class="v-toolbar elevation-0 theme--light transparent"
            style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
          >
            <div
              class="v-toolbar__content"
              style="height: 64px;"
            >
              <div
                class="v-responsive v-image"
                style="height: 64px;"
              >
                <div
                  class="v-image__image v-image__image--preload v-image__image--contain"
                  mode="in-out"
                  name="fade-transition"
                  style="background-position: center center;"
                />
                <div
                  class="v-responsive__content"
                />
              </div>
            </div>
          </nav>
           
          <div
            class="v-card__title justify-center v-card__title--primary"
          >
            <h3
              class="headline mb-0 primary--text font-weight-bold"
              style="justify-content: center;"
            >
              
        Log in to FoodFleet
      
            </h3>
          </div>
           
          <div
            class="v-card__text"
          >
            <div
              class="v-input v-text-field v-text-field--placeholder theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      aria-hidden="true"
                      class="v-label v-label--active theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      Email address
                    </label>
                    <input
                      aria-label="Email address"
                      name="login"
                      placeholder="Email address"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-text-field v-text-field--placeholder theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="password"
                      style="left: 0px; position: absolute;"
                    >
                      Password
                    </label>
                    <input
                      id="password"
                      name="password"
                      placeholder="Password"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="v-card__actions"
          >
            <div
              class="layout row wrap"
            >
              <div
                class="flex xs12"
              >
                
          Forgot password ?
          
                <button
                  class="text-uppercase v-btn v-btn--flat v-btn--small theme--light primary--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            Reset password here
          
                  </div>
                </button>
              </div>
               
              <div
                class="flex xs12"
              >
                <button
                  class="ff-login__button mt-2 mb-2 v-btn theme--light primary"
                  type="submit"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            Login
          
                  </div>
                </button>
              </div>
               
              <div
                class="flex"
              >
                <p
                  class="mb-1 mt-2"
                >
                  
            Not a member ?
          
                </p>
                 
                <p>
                  
            Sign up as a
            
                  <a
                    href="#register-customer"
                  >
                    Customer
                  </a>
                  
            or
            
                  <a
                    href="#register-supplier"
                  >
                    Join our fleet
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
         
      </form>
    </div>
  </div>
</div>
`;

exports[`components/users/Login Snapshots Loading 1`] = `
<div
  class="container fluid"
>
  <div
    class="layout align-center justify-center"
  >
    <div
      class="flex xs12 sm10 md4"
    >
      <form
        class="v-form"
        novalidate="novalidate"
      >
        <div
          class="elevation-12 rounded-card v-card v-sheet theme--light"
        >
          <nav
            class="v-toolbar elevation-0 theme--light transparent"
            style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
          >
            <div
              class="v-toolbar__content"
              style="height: 64px;"
            >
              <div
                class="v-responsive v-image"
                style="height: 64px;"
              >
                <div
                  class="v-image__image v-image__image--preload v-image__image--contain"
                  mode="in-out"
                  name="fade-transition"
                  style="background-position: center center;"
                />
                <div
                  class="v-responsive__content"
                />
              </div>
            </div>
          </nav>
           
          <div
            class="v-card__title justify-center v-card__title--primary"
          >
            <h3
              class="headline mb-0 primary--text font-weight-bold"
              style="justify-content: center;"
            >
              
        Log in to Foodfleet
      
            </h3>
          </div>
           
          <div
            class="v-card__text"
          >
            <div
              class="v-input v-text-field v-text-field--placeholder theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      aria-hidden="true"
                      class="v-label v-label--active theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      Email address
                    </label>
                    <input
                      aria-label="Email address"
                      name="login"
                      placeholder="Email address"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-text-field v-text-field--placeholder theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="password"
                      style="left: 0px; position: absolute;"
                    >
                      Password
                    </label>
                    <input
                      id="password"
                      name="password"
                      placeholder="Password"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="v-card__actions"
          >
            <div
              class="layout row wrap"
            >
              <div
                class="flex xs12"
              >
                
          Forgot password ?
          
                <button
                  class="text-uppercase v-btn v-btn--flat v-btn--small theme--light primary--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            Reset password here
          
                  </div>
                </button>
              </div>
               
              <div
                class="flex xs12"
              >
                <button
                  class="ff-login__button mt-2 mb-2 v-btn v-btn--loader theme--light primary"
                  type="submit"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            Login
          
                  </div>
                  <span
                    class="v-btn__loading"
                  >
                    <div
                      aria-valuemax="100"
                      aria-valuemin="0"
                      class="v-progress-circular v-progress-circular--indeterminate"
                      role="progressbar"
                      style="height: 23px; width: 23px;"
                    >
                      <svg
                        style="transform: rotate(0deg);"
                        viewBox="21.904761904761905 21.904761904761905 43.80952380952381 43.80952380952381"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          class="v-progress-circular__overlay"
                          cx="43.80952380952381"
                          cy="43.80952380952381"
                          fill="transparent"
                          r="20"
                          stroke-dasharray="125.664"
                          stroke-dashoffset="125.66370614359172px"
                          stroke-width="3.8095238095238093"
                        />
                      </svg>
                      <div
                        class="v-progress-circular__info"
                      />
                    </div>
                  </span>
                </button>
              </div>
               
              <div
                class="flex"
              >
                <p
                  class="mb-1 mt-2"
                >
                  
            Not a member ?
          
                </p>
                 
                <p>
                  
            Sign up as a
            
                  <a
                    href="#register-customer"
                  >
                    Customer
                  </a>
                  
            or
            
                  <a
                    href="#register-supplier"
                  >
                    Join our fleet
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
         
      </form>
    </div>
  </div>
</div>
`;

exports[`components/users/Login Snapshots WithError 1`] = `
<div
  class="container fluid"
>
  <div
    class="layout align-center justify-center"
  >
    <div
      class="flex xs12 sm10 md4"
    >
      <form
        class="v-form"
        novalidate="novalidate"
      >
        <div
          class="elevation-12 rounded-card v-card v-sheet theme--light"
        >
          <nav
            class="v-toolbar elevation-0 theme--light transparent"
            style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
          >
            <div
              class="v-toolbar__content"
              style="height: 64px;"
            >
              <div
                class="v-responsive v-image"
                style="height: 64px;"
              >
                <div
                  class="v-image__image v-image__image--preload v-image__image--contain"
                  mode="in-out"
                  name="fade-transition"
                  style="background-position: center center;"
                />
                <div
                  class="v-responsive__content"
                />
              </div>
            </div>
          </nav>
           
          <div
            class="v-card__title justify-center v-card__title--primary"
          >
            <h3
              class="headline mb-0 primary--text font-weight-bold"
              style="justify-content: center;"
            >
              
        Log in to FoodFleet
      
            </h3>
          </div>
           
          <div
            class="v-card__text"
          >
            <div
              class="v-input v-text-field v-text-field--placeholder theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      aria-hidden="true"
                      class="v-label v-label--active theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      Email address
                    </label>
                    <input
                      aria-label="Email address"
                      name="login"
                      placeholder="Email address"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-input v-text-field v-text-field--placeholder theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="password"
                      style="left: 0px; position: absolute;"
                    >
                      Password
                    </label>
                    <input
                      id="password"
                      name="password"
                      placeholder="Password"
                      type="password"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="v-card__actions"
          >
            <div
              class="layout row wrap"
            >
              <div
                class="flex xs12"
              >
                
          Forgot password ?
          
                <button
                  class="text-uppercase v-btn v-btn--flat v-btn--small theme--light primary--text"
                  type="button"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            Reset password here
          
                  </div>
                </button>
              </div>
               
              <div
                class="flex xs12"
              >
                <button
                  class="ff-login__button mt-2 mb-2 v-btn theme--light primary"
                  type="submit"
                >
                  <div
                    class="v-btn__content"
                  >
                    
            Login
          
                  </div>
                </button>
              </div>
               
              <div
                class="flex"
              >
                <p
                  class="mb-1 mt-2"
                >
                  
            Not a member ?
          
                </p>
                 
                <p>
                  
            Sign up as a
            
                  <a
                    href="#register-customer"
                  >
                    Customer
                  </a>
                  
            or
            
                  <a
                    href="#register-supplier"
                  >
                    Join our fleet
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="v-snack v-snack--active v-snack--top"
          name="v-snack-transition"
        >
          <div
            class="v-snack__wrapper error"
          >
            <div
              class="v-snack__content"
            >
              
              Wrong login or password
              
              <button
                class="v-btn v-btn--flat theme--dark"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
                Close
              
                </div>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
